src/modules/search/components/result/SubscriptionForm.tsx

```diff
@@ -0,0 +1,103 @@
+import React, { useEffect, useRef, useState } from 'react';
+import { Button, Checkbox, Notification, TextInput } from 'hds-react';
+import { useTranslation } from 'react-i18next';
+import styles from './SubscriptionForm.module.scss';
+import axios from 'axios';
+import { Project } from '../../../../types/common';
+import useSearchParams from '../../../../hooks/useSearchParams';
+
+type Props = {
+  onClose?: () => void;
+  project: Project;
+};
+
+enum FormState {
+  InProgress,
+  Error,
+  Ready,
+}
+
+const SubscriptionForm = ({ onClose, project }: Props) => {
+  const [email, setEmail] = useState('');
+  const [subscribeToNewsLetter, setSubscribeToNewsLetter] = useState(false);
+  const [state, setState] = useState(FormState.InProgress);
```

Comment:
Could we use some other key here not to confuse with oldschool state of class components?